spec/services/update_activity_as_project_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "rails_helper"
+
+RSpec.describe UpdateActivityAsProject do
+  let(:beis) { create(:beis_organisation) }
+  let(:parent) { create(:programme_activity) }
+  let(:activity) { create(:project_activity) }
+
+  describe "#call" do
+    let(:result) {
+      described_class.new(activity: activity, parent_id: parent.id).call
+    }
+
+    context "when the organisation is a Government organisation" do
+      it "uses the service owner as the reporting organisation" do
+        expect(result.reporting_organisation).to eq(beis)
+      end
+    end
+
+    context "when the organisation is a non-governmental organisation" do
+      let(:activity) { create(:project_activity, organisation: ngo) }
+      let(:ngo) { create(:delivery_partner_organisation, organisation_type: 21) }
+
+      it "saves the reporting organisation" do
```

Comment:
I like this version better "sets the reporting organisation to be the delivery partner"

Reply:
Agreed